[DOCPR-1022] Document manual workflow (#319)
canonical/sphinx-docs-starter-pack: docs/content/edit.rst

.. _edit:

Edit content
============

The landing page is stored in the :file:`docs/index.rst` file 

> while the rest of the pages are stored in the :file:`docs/content/` folder.

Reviewer: ```suggestion
while the rest of the pages are stored in the :file:`docs/content/` folder by default.
```

> The Navigation Menu structure is set by the ``.. toctree::`` blocks used to list all nested pages for a page. 

Reviewer: ```suggestion
The Navigation Menu structure is set by ``.. toctree::`` [directives](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-toctree). These directives define included content and the hierarchy of included content throughout the documentation.
```

The :file:`index.rst` page's ``toctree`` block contains the top level Navigation Menu.

To add a new page to the documentation:    

1. Create a new file in the `docs/content` folder. For example, to create the **Reference** page, create a document called :file:`reference.rst`, insert the following |RST|-formatted heading ``Reference`` at the beginning, and then save the file:

   .. code-block:: rest
      :caption: reStructuredText title example

         Reference
         =========

   If you prefer to use Markdown (MyST) syntax instead of |RST|, you can create a Markdown file. For example, :file:`reference.md` file with the following Markdown-formatted heading at the beginning:

   .. code-block:: markdown
      :caption: Markdown title example
         
         # Reference

2. Add the new file to the Navigation Menu. Open the :file:`index.rst` file or another page that you want to add a nested page to. At the bottom of the file locate the ``toctree`` block and add a properly indented line containing the path (without a file extension) to the new file created in the first step. For example, ``content/reference``.

   The ``toctree`` block will now look like this:

>    .. code-block:: rest
>          
>          .. toctree::
>          :hidden:
>          :maxdepth: 2
>          /content/quickstart
>          /content/setup
>          /content/update
>          /content/automatic_checks
>          /content/examples
>          /content/contributing
>          /content/reference

Reviewer: ```suggestion
   .. code-block:: rest
         
         .. toctree::
            :hidden:
            :maxdepth: 2

            /content/quickstart
            /content/setup
            /content/update
            /content/automatic_checks
            /content/examples
            /content/contributing
            /content/reference
```

Indentation and separation of options from the content of the directive are very important.

> The documentation will now show the new page added to the navigation.

Reviewer: ```suggestion
The documentation will now show the new page added to the navigation when rebuilt.
```

> By default, the page's title (the first heading in the file) is used for the Navigation Menu entry. You can overwrite a name of a Menu element by specifying it explicitly in the ``toctree`` block, for example: `Reference </content/reference>`.

Reviewer: ```suggestion
By default, the page's title (the first heading in the file) is used for the Navigation Menu entry. You can overwrite a name of a Menu element by specifying it explicitly in the ``toctree`` block, for example: `Reference </content/reference>`.

.. note::

   Sphinx based documentation is intended to contain only a single H1 heading in a single file.
```

Reviewer: Fair.
